src/main/java/org/threadly/concurrent/future/SettableListenableFuture.java

```diff
@@ -187,6 +241,8 @@ public T get() throws InterruptedException, ExecutionException {
       
       if (failure != null) {
         throw new ExecutionException(failure);
```

Comment:
would it at all make sence here to just see the CancellationException as just the ExecutionExceptions failure? and just set failure to "new CancellationException()"?